src/Workspaces/Core/Portable/Execution/SerializerService.cs

```diff
@@ -29,15 +30,8 @@ internal partial class Serializer
 
         private readonly ConcurrentDictionary<string, IOptionsSerializationService> _lazyLanguageSerializationService;
 
-        public Serializer(Solution solution) : this(solution.Workspace)
-        {
-        }
-
-        public Serializer(Workspace workspace) : this(workspace.Services)
-        {
-        }
-
-        public Serializer(HostWorkspaceServices workspaceServices)
+        [Obsolete("This exported object must be obtained through the MEF export provider.", error: true)]
+        public SerializerService(HostWorkspaceServices workspaceServices)
```

Comment:
cute.  i like that.  we should do that in a lot of places :)

Reply:
➡️ #25560